plan/plan.go

```diff
@@ -128,6 +128,8 @@ type LogicalPlan interface {
 
 	// buildKeyInfo will collect the information of unique keys into schema.
 	buildKeyInfo()
+
+	pushDownTopN(topn *Sort) LogicalPlan
```

Comment:
This interface contains a specific type in the argument, which is an anti-pattern.